File: src/app/conjugation/forms/causative-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {NonPastShortNegative} from "./non-past-short-negative";
import {CausativeAffirmative} from "./causative-affirmative";

export class CausativeNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {

        const originalType = word.wordType;
        const causative = new CausativeAffirmative().getConjugation(word);
        if (causative === undefined) {
            return undefined;
        }

        return new NonPastShortNegative().getConjugation(causative.changeType(WordType.IchidanVerb))?.changeType(originalType)
    }

    getTitle(): string {
        return "Causative, negative"
    }

    getSettingsTitle(): string {
        return "Negative"
    }

}

File: backend/app/conjugation/forms/causative_negative.py
from ..core import Conjugation, Word, WordType
from .causative_affirmative import CausativeAffirmative
from .non_past_short_negative import NonPastShortNegative


class CausativeNegative(Conjugation):

    title = 'Causative, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        original_type = word.word_type
        causative = CausativeAffirmative().conjugate(word)
        if causative is None:
            return None

        negative = NonPastShortNegative().conjugate(
            causative.change_type(WordType.ICHIDAN_VERB)
        )
        if negative is None:
            return None

        return negative.change_type(original_type)
